addons/mail/static/src/models/thread_view/thread_view.js

```diff
@@ -217,7 +227,12 @@ function factory(dependencies) {
          * Most recent message in this ThreadView that has been shown to the
          * current partner.
          */
-        lastVisibleMessage: many2one('mail.message'),
+        lastVisibleMessageOfThreadCache: many2one('mail.message', {
```

Comment:
Actually I don't even like the new name.

This implies the "last visible message" is a thread cache notion, except it isn't the case, it's purely a view notion that is correctly computed here.

I think `lastVisibleMessage` was a better name. It's obvious in this context of thread view this is always referencing it's currently displayed thread cache.

Reply:
So for this reason + change in stable + minimal diff, I think name should be reverted.

Reply:
@alexkuhn > @seb-odoo do not agree with your suggestion to change that field name, do I revert or can you convince him ?

Reply:
`lastVisibleMessageWithThreadCache`

Reply:
> I think `lastVisibleMessage` was a better name. It's obvious in this context of thread view this is always referencing it's currently displayed thread cache.

It's not obvious, notably it didn't make sense to reset this field on thread cache change. That's why `lastVisibleMessage` needs renaming so that it's more explicitly contextual to thread at hand.

Reply:
@seb-odoo  > "La parole est à la défense"

Reply:
We shouldn't do rename of this field in stable if the rename is not necessary for the fix to work. For example on the transient PR it made sense to use another name because we changed the meaning of the field. In this case we don't want to change its meaning, the rename would just clarify its current meaning. The fix is to update the field correctly, not to change its meaning.

As for what to do in master, I would suggest the field be renamed with a `threadCache` prefix because that's the convention in that model for other such fields (and if we don't like that convention we can reconsider and rename all others). Note that whatever is done for this field should also be done for other thread cache fields in this thread view model, including `checkedMessages`, `isLoading`, `lastMessage` (should have "thread" not cache in name), `lastVisibleMessage`, `messages`, `uncheckedMessages`.

Reply:
> [...] on the transient PR it made sense to use another name because we changed the meaning of the field. In this case we don't want to change its meaning, the rename would just clarify its current meaning. The fix is to update the field correctly, not to change its meaning.

Motivation of field name change with transient PR is exactly the same as here: name is too loose with how we use the field, and the root cause is field name being not clear enough. This is the same here. Both cases are not drastic change of meaning: they are renamed to be more focused on what they currently do and what they mean, to avoid confusion.

The rename of fields is a fix, because without rename the field is badly named. This is especially true in the context of this PR. I wouldn't have mind not renaming the field without this PR, but with this PR this is self-evident.

> As for what to do in master, I would suggest the field be renamed with a `threadCache` prefix because that's the convention in that model for other such fields (and if we don't like that convention we can reconsider and rename all others). Note that whatever is done for this field should also be done for other thread cache fields in this thread view model, including `checkedMessages`, `isLoading`, `lastMessage` (should have "thread" not cache in name), `lastVisibleMessage`, `messages`, `uncheckedMessages`.

Agree that fields that are contextual to thread cache should have it explicit in name. Not super fond of suggested convention to prefix with `threadCache`, because will make code more cumbersome to read and understand what such fields do just by their name, but it's more acceptable having it than none like at present.

I haven't given too much thought, but maybe modeling could have some improvements to make it more obvious for these fields to have a lifetime dictate by thread cache at hand? Something like having a sub-model of `mail.thread_view` that is recreate on change of targeted thread cache? Basically would solve long naming of such fields and simplify the somewhat "short-lived" aspect of these fields (and also group all of them together).

Reply:
I'm not comfortable with renaming this field (or any other field) in stable. I've seen other fixes use it and it can cause issues if we are not very careful. If we least we knew what to name it, I would suggest to create the new field and keep this one as a deprecated related field to the other (for stable reasons) but we don't even agree on what would be its ideal new name. Just add comment to field to clarify it's scope, it should be more than enough to avoid issue for now. I'm still open to reconsider this name on master of course.

And actually about last paragraph this might be the solution, this is the same "problem" for scroll position btw of which we stored in the view with thread cache as key (and I already expressed that for me whenever a field is storing an array or an object, it should probably be a relation instead) so I guess making an intermediary model for those would indeed make sense.

Reply:
Ok for not changing name in 14.0, adding context of use as comment of field, and making change of field name in master. Let's aim for very minimal changes in stable versions ;)